plugins/techdocs/src/components/DocsResultListItem/DocsResultListItem.tsx

```diff
@@ -32,29 +32,51 @@ const useStyles = makeStyles({
 export const DocsResultListItem = ({
   result,
   lineClamp = 5,
+  asListItem = true,
+  asLink = true,
 }: {
   result: any;
   lineClamp?: number;
+  asListItem?: boolean;
+  asLink?: boolean;
 }) => {
   const classes = useStyles();
-  return (
-    <Link to={result.location}>
-      <ListItem alignItems="flex-start" className={classes.flexContainer}>
-        <ListItemText
-          className={classes.itemText}
-          primaryTypographyProps={{ variant: 'h6' }}
-          primary={`${result.title} | ${result.name} docs `}
-          secondary={
-            <TextTruncate
-              line={lineClamp}
-              truncateText="…"
-              text={result.text}
-              element="span"
-            />
-          }
+  const TextItem = () => (
+    <ListItemText
+      className={classes.itemText}
+      primaryTypographyProps={{ variant: 'h6' }}
+      primary={`${result.title} | ${result.name} docs `}
```

Comment:
Hmm...  I suppose in the context of a single entity, the `${result.name} docs` is also a bit redundant.

I feel a little hesitant about putting so much conditional logic in here just to satisfy the in-context use-case, but the benefits of sharing the implementation are high enough to justify it, in my opinion.

Reply:
Yeah. it's a getting to that awkward state where it's kinda the generic but in reality directed so much with props that it's not really. I'll add one more conditional here.